src/plugins/memdump/userhook.cpp

```diff
@@ -122,21 +122,219 @@
+}
+
+static event_response_t usermode_return_hook_cb(drakvuf_t drakvuf, drakvuf_trap_info* info)
+{
+    return_hook_target_entry_t* ret_target = (return_hook_target_entry_t*)info->trap->data;
+    if (info->proc_data.pid != ret_target->pid)
+        return VMI_EVENT_RESPONSE_NONE;
+
+    auto plugin = ret_target->plugin;
+
+    gchar* escaped_pname;
+    switch (plugin->m_output_format)
+    {
+        case OUTPUT_CSV:
+            printf("memdump-userhok," FORMAT_TIMEVAL ",%" PRIu32 ",0x%" PRIx64 ",\"%s\",%" PRIi64 ",\"%s\",0x%" PRIx64 ",\"",
+                   UNPACK_TIMEVAL(info->timestamp), info->vcpu, info->regs->cr3, info->proc_data.name,
+                   info->proc_data.userid, info->trap->name, info->regs->rip);
+            print_arguments(ret_target->arguments);
+            printf("\"");
+            break;
+        case OUTPUT_KV:
+            printf("memdump, Time=" FORMAT_TIMEVAL ",VCPU=%" PRIu32 ",CR3=0x%" PRIx64 ",ProcessName=\"%s\",UserID=%" PRIi64 ",Method=\"%s\",CalledFrom=0x%" PRIx64 ",Arguments=\"",
+                   UNPACK_TIMEVAL(info->timestamp), info->vcpu, info->regs->cr3, info->proc_data.name,
+                   info->proc_data.userid, info->trap->name, info->regs->rip);
+            print_arguments(ret_target->arguments);
+            printf("\"");
+            break;
+        case OUTPUT_JSON:
+            escaped_pname = drakvuf_escape_str(info->proc_data.name);
+            printf( "{"
+                    "\"Plugin\": \"memdump-userhook\", "
+                    "\"TimeStamp\":" "\"" FORMAT_TIMEVAL "\", "
+                    "\"ProcessName\": %s, "
+                    "\"UserName\": \"%s\", "
+                    "\"UserId\": %" PRIu64 ", "
+                    "\"PID\": %d, "
+                    "\"PPID\": %d, "
+                    "\"Method\": \"%s\", "
+                    "\"CalledFrom\": 0x%" PRIx64 ", "
+                    "\"Arguments\": [",
+                    UNPACK_TIMEVAL(info->timestamp),
+                    escaped_pname,
+                    USERIDSTR(drakvuf), info->proc_data.userid,
+                    info->proc_data.pid, info->proc_data.ppid,
+                    info->trap->name,
+                    info->regs->rip);
+
+            print_arguments(ret_target->arguments);
+            printf("}");
+            g_free(escaped_pname);
+            break;
+        default:
+        case OUTPUT_DEFAULT:
+            printf("[MEMDUMP-USERHOOK] TIME:" FORMAT_TIMEVAL " VCPU:%" PRIu32 " CR3:0x%" PRIx64 " ProcessName:\"%s\" UserID:%" PRIi64 " Method:\"%s\" CalledFrom:0x%" PRIx64 " Arguments:\"",
+                   UNPACK_TIMEVAL(info->timestamp), info->vcpu, info->regs->cr3, info->proc_data.name,
+                   info->proc_data.userid, info->trap->name, info->regs->rip);
+            print_arguments(ret_target->arguments);
+            printf("\"");
+            break;
+    }
+    printf("\n");
+
+    drakvuf_remove_trap(drakvuf, info->trap, nullptr);
+	delete ret_target;
```

Comment:
Alignment looks off here as well and a couple more spots. Could you go through the PR and make sure you are not mixing in tabs instead of spaces?

Reply:
my IDE was clearly drunk mixing tabs and spaces... fixed

Reply:
btw. seems like apt repository has some random outage
https://travis-ci.org/tklengyel/drakvuf/jobs/625664390?utm_medium=notification&utm_source=github_status